src/python/pants/backend/docker/util_rules/docker_binary.py

```diff
@@ -37,7 +37,9 @@ def build_image(
         env: Mapping[str, str] | None = None,
         extra_args: tuple[str, ...] = (),
     ) -> Process:
-        args = [self.path, "build", *extra_args]
+        # If the first extra arg is "build", we use `docker buildx build ...`
+        subcommand = "buildx" if "build" in extra_args[:1] else "build"
```

Comment:
I've not considered if it would be better with a more explicit way to signal the use `docker buildx` rather than passing in `build` as an extra arg...

Reply:
Buildx is required for the `--push` (or `--output=type=registry`) option.